django/core/management/commands/shell.py

```diff
@@ -39,36 +44,47 @@ def add_arguments(self, parser):
 
     def _ipython_pre_011(self):
         """Start IPython pre-0.11"""
-        from IPython.Shell import IPShell
+        try:
+            from IPython.Shell import IPShell
+        except ImportError:
+            raise ShellImportError
```

Comment:
I think it's preferable to raise instances rather than classes. Python 3 will instantiate it for you when required anyway ([docs](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement)):

> Otherwise, raise evaluates the first expression as the exception object. It must be either a subclass or an instance of BaseException. If it is a class, the exception instance will be obtained when needed by instantiating the class with no arguments.